Changes.md

```diff
@@ -2,6 +2,24 @@
 
 ## H2O
 
+### 3.46.0.3 - 6/10/2024
```

Comment:
Maybe 7/10 and add https://github.com/h2oai/h2o-3/issues/15810 if its going to be part of the release.